### PR TITLE
sectorbuilder: Check free space before creating sectors

### PR DESCRIPTION
**Pull request by magik6k**
This also includes some initial refactoring in preparation for jbod support

### CODE REVIEW

File: chain/types/bigint.go [hunk 1]
```diff
@@ -124,6 +124,22 @@ func (bi *BigInt) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+var sizeUnits = []string{"B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB"}
```
Comment (whyrusleeping) [inline chain/types/bigint.go]: if we're just copying this from somewhere, can we reuse it?

Comment (whyrusleeping) [reply chain/types/bigint.go]: oh nvm, i see. we're moving it to here

File: lib/sectorbuilder/fs.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	return out
+}
+
+func (f *fs) reserve(typ dataType, size uint64) error {
+	f.lk.Lock()
+	defer f.lk.Unlock()
+
+	var fsstat syscall.Statfs_t
+
+	if err := syscall.Statfs(f.path, &fsstat); err != nil {
```
Comment (whyrusleeping) [inline lib/sectorbuilder/fs.go]: lets stat the subdirectory, on my computer I map the staged and cache directory to my SSD, and the sealed to an nfs mount

File: lib/sectorbuilder/sectorbuilder.go [hunk 3]
```diff
@@ -433,6 +428,18 @@ func (sb *SectorBuilder) sealPreCommitRemote(call workerCall) (RawSealPreCommitO
 }
 
 func (sb *SectorBuilder) SealPreCommit(sectorID uint64, ticket SealTicket, pieces []PublicPieceInfo) (RawSealPreCommitOutput, error) {
+	fs := sb.filesystem
+
+	if err := fs.reserve(dataCache, sb.ssize); err != nil {
```
Comment (whyrusleeping) [inline lib/sectorbuilder/sectorbuilder.go]: isnt it like sb.ssize  * 10?

File: chain/types/bigint.go (hunk 1 above)
Comment (magik6k) [reply chain/types/bigint.go]: I moved this from `cli/utils.go`

File: lib/sectorbuilder/sectorbuilder.go (hunk 3 above)
Comment (magik6k) [reply lib/sectorbuilder/sectorbuilder.go]: See `fs.go:overheadMul`

**Pull request merged**